library/HTMLPurifier/Filter/URIFilter/SafeLink.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+class HTMLPurifier_URIFilter_SafeLink extends HTMLPurifier_URIFilter
+{
+    /**
+     * @type string
+     */
+    public $name = 'SafeLink';
+
+    /**
+     * @type bool
+     */
+    public $always_load = true;
+
+    /**
+     * @type string
+     */
+    protected $regexp = null;
+
+    /**
+     * @param HTMLPurifier_Config $config
+     * @return bool
+     */
+    public function prepare($config)
+    {
+        $this->regexp = $config->get('URI.SafeLinkRegexp');
+
+        return true;
+    }
+
+    /**
+     * @param HTMLPurifier_URI $uri
+     * @param HTMLPurifier_Config $config
+     * @param HTMLPurifier_Context $context
+     * @return bool
+     */
+    public function filter(&$uri, $config, $context)
+    {
+        // check if the filter should actually trigger
+        if (!$context->get('EmbeddedURI', true)) {
+            return true;
+        }
+
+        $token = $context->get('CurrentToken', true);
+        if (! ($token && $token->name == 'link')) {
```

Comment:
I don't like spaces after negation operator 😁

Reply:
This was copied from HTMLPurifier 😅  will change it :)